scripts/validation_shell.js

```diff
@@ -0,0 +1,40 @@
+// Test a file is valid, by just loading it.
+
+// Shell integration.
+if (typeof console === 'undefined') {
+  console = { log: print };
+}
+var tempRet0;
+var binary;
+if (typeof process === 'object' && typeof require === 'function' /* node.js detection */) {
+  var args = process.argv.slice(2);
+  binary = require('fs').readFileSync(args[0]);
+  if (!binary.buffer) binary = new Uint8Array(binary);
+} else {
+  var args;
+  if (typeof scriptArgs != 'undefined') {
+    args = scriptArgs;
+  } else if (typeof arguments != 'undefined') {
+    args = arguments;
+  }
+  if (typeof readbuffer === 'function') {
+    binary = new Uint8Array(readbuffer(args[0]));
+  } else {
+    binary = read(args[0], 'binary');
+  }
+}
+
+// Test the wasm for validity. The imports aren't there (we can't use a proxy
+// because we don't know the types), but if we get to the imports, that means
+// the wasm is valid.
```

Comment:
Don't know much about the v8 validation process. Are imports validated after all other things?

Reply:
Yes, it compiles first, then uses imports only when creating an instance (at least v8 and sm do - perhaps chakra does lazy validation/compilation?).

Thinking about this again now, I see the code can be much simpler - we can just compile and not even create an instance, so no imports are involved.